test/forms/behaviours/StringFieldBehaviours.scala

```diff
@@ -31,6 +31,16 @@ trait StringFieldBehaviours extends FieldBehaviours {
       }
     }
 
+  def fieldWithMaxLengthPhoneNumber(form: Form[_], fieldName: String, maxLength: Int, lengthError: FormError): Unit =
```

Comment:
Is this needed? Looks to provide the same function as fieldWithMaxLength

Reply:
it uses a different generator specific to phone numbers. fieldWithMaxLength can generate anything from symbols to letters etc so because the field can only contain numbers this was needed.

Reply:
In fieldWithMaxLength test will check within the sequence of errors for the length error, regardless if there are other validation errors there or not - whereas this (fieldWithMaxLengthPhoneNumber) looks for it by satisfying other validation.

I suppose what i mean is that this test is coupled to the phone number formatting where what we want to test here is just the length

Reply:
(continued elsewhere)